test_conformance/spirv_new/test_decorate.cpp

```diff
@@ -389,41 +461,62 @@ template<typename Ti, typename To>
 }
 
 template<typename Ti, typename To>
 inline To round_to_posinf(Ti in)
 {
-    To out = std::ceil(in);
+    To out;
+    if (std::is_same<Ti, cl_half>::value)
+        out = std::ceil(cl_half_to_float(in));
+    else
+        out = std::ceil(in);
     return out;
 }
 
 template<typename Ti, typename To>
 inline To round_to_neginf(Ti in)
 {
-    To out = std::floor(in);
+    To out;
+    if (std::is_same<Ti, cl_half>::value)
+        out = std::floor(cl_half_to_float(in));
+    else
+        out = std::floor(in);
     return out;
 }
 
-#define TEST_SPIRV_FP_ROUNDING_DECORATE(name, func, Ti, To)             \
-    TEST_SPIRV_FUNC(decorate_fp_rounding_mode_##name##_##Ti##_##To)     \
-    {                                                                   \
-        typedef cl_##Ti clTi;                                           \
-        typedef cl_##To clTo;                                           \
-        const int num = 1 << 16;                                        \
-        std::vector<clTi> in(num);                                      \
-        std::vector<clTo>  out(num);                                    \
-        RandomSeed seed(gRandomSeed);                                   \
-                                                                        \
-        for (int i = 0; i < num; i++) {                                 \
-            in[i] = num * genrand<clTi>(seed) - num/2;                  \
-            out[i] = func<clTi, clTo>(in[i]);                           \
-        }                                                               \
-        const char *name = "decorate_rounding_" #name "_" #Ti "_" #To;  \
-        return test_fp_rounding(deviceID, context, queue,               \
-                                name, in, out);                         \
-    }                                                                   \
+#define TEST_SPIRV_FP_ROUNDING_DECORATE(name, func, Ti, To)                    \
+    TEST_SPIRV_FUNC(decorate_fp_rounding_mode_##name##_##Ti##_##To)            \
+    {                                                                          \
+        typedef cl_##Ti clTi;                                                  \
+        typedef cl_##To clTo;                                                  \
+        const int num = 1 << 16;                                               \
+        std::vector<clTi> in(num);                                             \
+        std::vector<clTo> out(num);                                            \
+        RandomSeed seed(gRandomSeed);                                          \
+                                                                               \
+        for (int i = 0; i < num; i++)                                          \
+        {                                                                      \
+            in[i] = num * genrand<clTi>(seed) - num / 2;                       \
```

Comment:
Is there a possibility with the above code we could overflow half float with the rounding but the float result  we calculate would not overflow, so the results don't really match? Do we need to constrain the inputs more?

Reply:
is it possible to comment on this points before I approve?

Reply:
I took a look at this yesterday and it's actually a little worse than I expected: `genrand<cl_half>(seed)` returns a `short`, not a floating-point value between zero and one, so almost all of these values overflow and we really only test one value.  This needs to be fixed.